addons/pkg/util/tkr_util_test.go

```diff
@@ -138,14 +142,74 @@ var _ = Describe("TKR utils", func() {
 			BeforeEach(func() {
 				crtCtl = &fakeclusterclient.CRTClusterClient{}
 				tkrName = testTKR
-				tkrV1Alpha3, err = GetTKRByNameV1Alpha3(ctx, crtCtl, tkrName)
+				tkrV1Alpha3, err = GetTKRByNameV1Alpha3(ctx, crtCtl, nil, tkrName)
 			})
 
 			It("should return nil TKR", func() {
 				Expect(err).ShouldNot(HaveOccurred())
 				Expect(tkrV1Alpha3).ShouldNot(BeNil())
 			})
 		})
+
+		When("tkr spec is available as annotations on cluster resource", func() {
```

Comment:
you call it "tkr spec" here but  just "tkr" in the next :) . The correct term should be tkr...
tkr spec sounds to me like the .spec of the TKR CR which is not what you mean